src/core/flate_stream.js

```diff
@@ -161,8 +161,19 @@ class FlateStream extends DecodeStream {
     try {
       const { readable, writable } = new DecompressionStream("deflate");
       const writer = writable.getWriter();
-      writer.write(bytes);
-      writer.close();
+      await writer.ready;
+
+      // We can't await writer.write() because it'll block until the reader
+      // starts which happens few lines below.
+      writer
+        .write(bytes)
+        .then(async () => {
+          try {
+            await writer.ready;
+            await writer.close();
+          } catch {}
+        })
+        .catch(() => {});
```

Comment:
Isn't the try-catch redundant here, since any error thrown there *should* already be caught by the catch-handler at the end!?
```suggestion
        .then(async () => {
          await writer.ready;
          await writer.close();
        })
        .catch(() => {});
```